src/web/app/components/new-instructor-data-row/new-instructor-data-row.component.spec.ts

```diff
@@ -0,0 +1,362 @@
+      .nativeElement.textContent;
+    expect(displayedInstructorEmail).toEqual(expectedInstructorData.email);
+  });
+
+  it('should display the instructor institution received as input', () => {
+    const displayedInstructorInstitution: string = fixture.debugElement
+      .query(By.css(`#instructor-${expectedIndex}-institution`))
+      .nativeElement.textContent;
+    expect(displayedInstructorInstitution).toEqual(expectedInstructorData.institution);
+  });
+
+  it('should have an add button that is not disabled when there are zero active requests as input', () => {
+    expect(addButtonEl.disabled).toBeFalsy();
+  });
+
+  it('should have an add button that is disabled when there are non-zero active requests as input', () => {
+    component.activeRequests = 1;
+    fixture.detectChanges();
+    expect(addButtonEl.disabled).toBeTruthy();
+  });
+
+  it('should emit addInstructorEvent when adding', () => {
+    let hasEmitted: boolean = false;
+    component.addInstructorEvent
+      .pipe(first())
+      .subscribe(() => hasEmitted = true);
+
+    addButtonDe.triggerEventHandler('click', null);
+    expect(hasEmitted).toBeTruthy();
+  });
+
+  it('should emit removeInstructorEvent when removing', () => {
+    let hasEmitted: boolean = false;
+    component.removeInstructorEvent
+      .pipe(first())
+      .subscribe(() => hasEmitted = true);
+
+    fixture.debugElement
+      .query(By.css(`#remove-instructor-${expectedIndex}`))
+      .triggerEventHandler('click', null);
+    expect(hasEmitted).toBeTruthy();
+  });
+
+  it('should emit true via toggleEditModeEvent when entering edit mode', () => {
+    let isInEditMode: boolean | undefined;
+    component.toggleEditModeEvent
+      .pipe(first())
+      .subscribe((isBeingEdited: boolean) => isInEditMode = isBeingEdited);
+
+    editButtonDe.triggerEventHandler('click', null);
+    expect(isInEditMode).toBeTruthy();
+  });
+
+  it('should emit false via toggleEditModeEvent when confirming the edit', () => {
+    let isInEditMode: boolean | undefined;
+    component.toggleEditModeEvent
+      .subscribe((isBeingEdited: boolean) => isInEditMode = isBeingEdited);
+
+    editButtonDe.triggerEventHandler('click', null);
+    fixture.detectChanges();
+
+    fixture.debugElement
+      .query(By.css(`#confirm-edit-instructor-${expectedIndex}`))
+      .triggerEventHandler('click', null);
+    expect(isInEditMode).toBeFalsy();
+  });
+
+  it('should emit false via toggleEditModeEvent when cancelling the edit', () => {
+    let isInEditMode: boolean | undefined;
+    component.toggleEditModeEvent
+      .subscribe((isBeingEdited: boolean) => isInEditMode = isBeingEdited);
+
+    editButtonDe.triggerEventHandler('click', null);
+    fixture.detectChanges();
+
+    fixture.debugElement
+      .query(By.css(`#cancel-edit-instructor-${expectedIndex}`))
+      .triggerEventHandler('click', null);
+    expect(isInEditMode).toBeFalsy();
+  });
+
+  it('should initialise isBeingEdited to false', () => {
```

Comment:
The two "should initialise" cases should be part of "should create", because they are related to when the component is freshly created.

Reply:
I see. I initially thought "should create" is only for whether or not the object is created, not what happens. I will change this.